### PR TITLE
Add Story 3 team access request flow

### PR DESCRIPTION
**Pull request by kevinrapley**
## Summary

Implements Story 3 for `ROPS-AUTH-P1-000 — Governed access, permissions and audit`.

This PR lets signed-in users request access to a team without approving access, assigning roles, creating teams, or granting team-scoped records while the request is pending.

## Story 3 scope

As a signed-in ResearchOps user without the team access I need, I need to request access to a team, so that a Team Admin can review my request before I can use team-scoped ResearchOps records.

## Team position

> Story 3 should let users ask to enter a team.
> It should not decide what they can do once they are inside.

## Changes

- Add Story 3 product plan.
- Add `auth_team_access_requests` D1 migration.
- Add route declarations for:
  - `GET /api/team-access/requests`
  - `POST /api/team-access/requests`
  - `POST /api/team-access/requests/cancel`
- Add Worker handler for listing, creating and cancelling team access requests.
- Add request-access GOV.UK source template.
- Add generated `/pages/account/team-access/` page.
- Add browser controller for the request-access form.
- Update the account page to show a separate `Team access requests` section.
- Update the account page no-team state to link to `Request access to a team`.
- Add route-state tests for the page, API route declarations, Worker wiring, D1 migration and scope boundaries.
- Add feature-branch trace artefacts.

## Scope controls

This PR does not add:

- Team Admin approval queue
- approve or reject request
- role request
- role assignment
- team switching
- team creation
- notification delivery
- PII access
- audit viewer

Pending requests do not create active team memberships and do not create role assignments.

## Data model

Adds:

- `auth_team_access_requests`

The request state is separate from `auth_team_memberships`, so pending access is not treated as active membership.

Initial request statuses covered by this story:

- `pending`
- `cancelled`

Future statuses reserved for later approval work:

- `approved`
- `rejected`

## Trace

- `docs/agent-audit/reasoning/2026/05/30/auth-story-3-team-access-request.md`
- `docs/agent-audit/reasoning/2026/05/30/auth-story-3-team-access-request.json`

## Changed-file verification

Changed files: 15

- `docs/agent-audit/reasoning/2026/05/30/auth-story-3-team-access-request.json`
- `docs/agent-audit/reasoning/2026/05/30/auth-story-3-team-access-request.md`
- `docs/product/26/05/30/auth-story-3-team-access-request.md`
- `infra/cloudflare/migrations/0005_auth_team_access_requests.sql`
- `infra/cloudflare/src/core/auth/team-access-requests.js`
- `infra/cloudflare/src/worker.js`
- `public/js/auth-account-page.js`
- `public/js/auth-team-access-page.js`
- `public/pages/account/index.html`
- `public/pages/account/team-access/index.html`
- `scripts/govuk/render-govuk-pages.mjs`
- `src/govuk/templates/pages/account-team-access.njk`
- `src/govuk/templates/pages/account.njk`
- `tests/auth-account-dashboard-route-state.test.js`
- `tests/auth-team-access-request-route-state.test.js`

Branch comparison: 19 commits ahead of `main`, 0 behind.

## Validation

Not run locally in this connector context.

Ready for GitHub checks to run:

- `npm run validate`
- `npm run lint`
- `npm test -- --ci`

## Manual browser checks

Open:

`/pages/account/`

Confirm:

- no-team state links to `Request access to a team`
- pending requests show separately from active team memberships
- pending requests are labelled `Awaiting approval`
- pending requests say they do not give access to team records yet
- cancellation removes a pending request from the account page

Open:

`/pages/account/team-access/`

Confirm:

- page title is `Request access to a team`
- form has a team name or invitation code field
- form has an optional Team Admin message field
- missing team reference shows an accessible error
- valid active team request creates a pending request
- duplicate pending request is not created
- active team member cannot request the same team again

## Known limitation

Invitation-code handling is not implemented as a separate secure invitation-code system in this first slice. The form accepts a team name, team ID or entered reference, but the backend currently resolves discoverable active team IDs or names only. A real invitation-code model should be a later slice because it needs expiry, lookup, anti-enumeration and audit rules.

### CODE REVIEW

File: scripts/govuk/render-govuk-pages.mjs [hunk 1]
```diff
@@ -127,10 +132,17 @@ const pages = [
...
-			navigation: navigation.map((item) => ({
-				...item,
-				active: false,
-			})),
+			navigation: accountNavigation,
+		},
+	},
+	{
+		template: 'pages/account-team-access.njk',
+		output: 'public/pages/account/team-access/index.html',
```
[the remark this answers is not in the document]
Comment (kevinrapley) [reply scripts/govuk/render-govuk-pages.mjs:140]: Resolved in the follow-up commits.

The new generated route `/pages/account/team-access/index.html` is now registered in `visual-walkthrough.config.mjs` as `account-team-access` in the Account group. The walkthrough registry contract was then corrected so existing route/state coverage remained intact.

Current PR head `ef8dd400248d0c845dba5934165c2f3d3d2f201d` is green across CI, Worker CI, Release Gate, Render GOV.UK pages, pa11y-ci, qa-bdd, Lychee, Format pull request and Validate ResearchOps.
